src/coreComponents/linearAlgebra/interfaces/hypre/HypreUtils.hpp

```diff
@@ -498,7 +498,8 @@ enum class MGRRestrictionType : HYPRE_Int
   jacobi = 2,              //!< Diagonal scaling
   approximateInverse = 3,  //!< Approximate inverse
   blockJacobi = 12,        //!< Block-Jacobi
-  cprLike = 13             //!< CPR-like restriction
+  cprLike = 13,            //!< CPR-like restriction
+  blockColsum = 14         //!< Block column sum approximation
```

Comment:
`cprLike=quasi-IMPES` and `blockColsum=TrueIMPES`?

Reply:
Correct, @castelletto1, should we update these names or add a comment?

Reply:
I agree that we need a better/coherent terminology. I don't think we should ever refer to `CPR`, `TrueIMPES` or `QuasiIMPES` here. These concepts are physics-related. Here we should stick as much as possible a mathematical terminology. We should find terms that express what type of sparsification is introduced in the restriction operator (and similarly for the prolongation) for the term $`-A_{CF} A_{FF}^{-1}`$.

In both QuasiIMPES and TrueIMPES a block diagonal approximation is introduced for both $`A_{CF}`$ and $`A_{FF}`$ - the goal being preserving the sparsity of $`A_{CC}`$ while computing the Schur complement.

Now, in QuasiIMPES, or at least in my understanding of it, we approximate $`A_{FF}`$ by the block diagonal, with size of each dense square block equal to the number of non pressure DOFs. Similarly, for $`A_{CF}`$ we keep the _main block diagonal_, where the block are row vectors having length equal to the size of $`A_{FF}`$'s blocks.

In TrueIMPES, we do something similar, but the diagonal block are obtained via column summation (block-wise). I would probably refer to this approach as `lumping`, with summation taking place over columns.

To conclude, an unambiguous definition of the approximation introduced for the ideal restriction operator would require the following information:
1. how we approximate $`A_{FF}`$
2. how we invert the approximate $`A_{FF}`$
3. how we approximate $`A_{CF}`$

For 1., we could have:
- diagonal approximation
- block diagonal approximation (this would be used for Quasi IMPES)
- lumped diagonal approximation summing rows or columns
- lumped block diagonal approximation summing blocks of rows or columns (this would be used for True IMPES)
- no approximation

As to 2., for the first four approximations above, we would use exact inversion. For the last option, we could use an approximate inverse or a direct solver.

For 3., I would consider the same approximation options listed for 1.

Reply:
Very nice comment. I guess ideally it would be great to have a single name referring to 1, 2, and 3 all at once. But that seems hard to me (any ideas?). 

Still, it might be a good idea to refer to True IMPES or quasi IMPES in a comment so that the reservoir simulation folks understand what we are talking about

Reply:
> Still, it might be a good idea to refer to True IMPES or quasi IMPES in a comment so that the reservoir simulation folks understand what we are talking about

Absolutely, the comment should be included in the appropriate MGR recipe.

Reply:
> Very nice comment. I guess ideally it would be great to have a single name referring to 1, 2, and 3 all at once. But that seems hard to me (any ideas?).

I think we can have a single name for 1 and 2. Every time we have a diagonal approximation (diagonal, block diagonal, with possible lumping over rows and columns), an exact inversion follows. If we chose an approximate inverse or a direct solver, it automatically implies that no previous sparsification is introduced. 

However, what is done for 3 is in principle independent of the choices done in 1.

Reply:
Do we anticipate any use case for the situation where what is done for 3 is different to what is done for 1?

I'm referring to what the new capability in this PR as `blockColSum`, but now I see `blockColLump` is probably more appropriate. Thoughts or different suggestions?

Reply:
For example, our fixed stress preconditioner uses a diagonal approximation for $`A_{FF}`$, but keeps the $`A_{CF}`$ as is. Then a further, approximation is introduced in a non variational sense (`nonGalekin`), by preserving only the diagonal of the correction term $`A_{CF} A_{FF}^{-1} A_{FC}`$.

Reply:
I would be ok with `blockColLumped` or similar term and an implied exact inversion and a more detailed comment in the recipe

Reply:
Thanks all, selected `blockColLumped`